recipe/build.sh

```diff
@@ -3,17 +3,19 @@ set -euxo pipefail
 
 rm -rf build || true
 
+
 if [ ${cuda_compiler_version} != "None" ]; then
   CUDA_CMAKE_OPTIONS="-DCMAKE_CUDA_COMPILER=${CUDA_HOME}/bin/nvcc -DCMAKE_CUDA_HOST_COMPILER=${CXX} -DCUDA_ARCH_NAME=All "
   USE_CUDA=ON
+  # Remove -std=c++17 from CXXFLAGS for compatibility with nvcc
+  export CXXFLAGS="$(echo $CXXFLAGS | sed -e 's/ -std=[^ ]*//')"
+  export CFLAGS="$(echo $CFLAGS | sed -e 's/ -mtune=[^ ]*//')"
 else
   CUDA_CMAKE_OPTIONS=""
   USE_CUDA=OFF
+  CXXFLAGS="${CXXFLAGS} --std=c++17 "
```

Comment:
Think it is only one `-`. Also we may want to specify C11

```suggestion
  CFLAGS="${CFLAGS} -std=c11 "
  CXXFLAGS="${CXXFLAGS} -std=c++17 "
```

Reply:
@jakirkham I think you posted this RIGHT as I found that mistake :laughing: 
![image](https://github.com/conda-forge/dgl-feedstock/assets/11765982/d9df998a-bf97-4057-9065-a70f226fe1e7)

I am going to see if CI will pass with 17, but 11 might be better. I tired running build local on my linux box and even with the sdk `OSX_SDK_DIR=/home/mmh/Downloads/MacOSX10.9.sdk` I get 

```
Mamba failed to solve:
 - liblapacke 3.9.* *netlib
 - dmlc 0.5.*
 - llvm-openmp >=15.0.7
 - pytorch 2.0.*
 - dlpack 0.7.*
 - metis 5.1.1.*
 - libcxx >=15.0.7
 - libblas 3.9.* *netlib
 - nanoflann
 - pytorch * *cpu*
 - _openmp_mutex >=4.5
 - python 3.9.* *_cpython
 - parallel-hashmap
 - _openmp_mutex * *_llvm
 - cython <3
 - gklib 5.1.1.*
 - liblapack 3.9.* *netlib
 - setuptools
 - libcblas 3.9.* *netlib

with channels:

The reported errors are:
- Encountered problems while solving:
-   - nothing provides requested _openmp_mutex >=4.5
-   - nothing provides requested _openmp_mutex * *_llvm
```

So I don't think I can build an osx package on a linux box

Reply:
Getting the same error with 17, I will try 11 but I think I might need to bump the osx sdk so it can use the newer `cstdlib.h`

Reply:
Oh and I will try it as a `CFLAGS` first!